common/lsp/lsp-protocol-operators.h

```diff
@@ -31,8 +31,12 @@ inline constexpr bool operator<(const Position &a, const Position &b) {
 inline constexpr bool operator>=(const Position &a, const Position &b) {
   return !(a < b);
 }
+inline constexpr bool operator==(const Position &a, const Position &b) {
+  return a.line == b.line && a.character == b.character;
+}
+
 inline constexpr bool rangeOverlap(const Range &a, const Range &b) {
-  return !(a.start >= b.end || b.start >= a.end);
+  return !(a.start >= b.end || b.start >= a.end) || (a.start == b.start);
```

Comment:
How about: `!(a.end <= b.start || b.end <= a.start)`
Overlap check should only require two comparisons at most.

Reply:
That expression would be equivalent to the original one (`a.end <= b.start` equiv to `b.start >= a.end` as well as `b.end <= a.start` equiv to `a.start >= b.end`) and will not catch the empty range.
It probably can be formulated in two expressions, but haven't had coffee yet. Do you see a way ?

Reply:
If there is a "special case", then you probably can't avoid the extra comparison.